### PR TITLE
Fixes #15377 - detect importer and dist changes on capsule

### PR DESCRIPTION
**Pull request by jlsherrill**
sync

### CODE REVIEW

**Comment by beav** [comment]
works as advertised, APJ

File: app/lib/actions/pulp/repository/refresh.rb [hunk 1]
```diff
@@ -15,22 +15,21 @@ def plan(repository, input = {})
 
         def update_or_associate_importer(capsule_id, repository, repository_details)
           existing_importers = repository_details["importers"]
-          importer = capsule_id ? repository.generate_importer(true) : repository.generate_importer
-          importer_config = capsule_id ? importer.config.merge!(importer_certs(repository)) : importer.config
+          importer = repository.generate_importer(!capsule_id.nil?)
```
Comment (johnpmitsch) [inline app/lib/actions/pulp/repository/refresh.rb]: where are the importer certs coming from now?

Comment (jlsherrill) [inline app/lib/actions/pulp/repository/refresh.rb]: The generate_importer method in repo.rb

Comment (johnpmitsch) [inline app/lib/actions/pulp/repository/refresh.rb]: :+1:

File: app/models/katello/glue/pulp/repo.rb [hunk 2]
```diff
@@ -128,15 +128,14 @@ def generate_importer(capsule = false)
         when Repository::FILE_TYPE
           Runcible::Models::IsoImporter.new(importer_ssl_options(capsule).merge(:feed => importer_feed_url(capsule)))
         when Repository::PUPPET_TYPE
-          options = {}
-          options[:feed] = importer_feed_url(capsule)
-          Runcible::Models::PuppetImporter.new(options)
+          options = {:feed => impoter_feed_url(capsule)}
```
Comment (johnpmitsch) [inline app/models/katello/glue/pulp/repo.rb]: should be importer_feed_url (missing the 'r')

**Comment by johnpmitsch** [comment]
@jlsherrill Once I fixed the typo, syncs are running correctly, and the distributors and importers are not updating after they have been synced (which is correct). I tested updating a yum repos config, and it updated the importer config.

**Comment by beav** [comment]
@johnpmitsch good catch, i totally missed that in my testing

**Comment by jlsherrill** [comment]
Good catch john, that was the only type i didn't fully test :)

updated

**Comment by johnpmitsch** [comment]
ACK

**Pull request merged**